File: Ex78ViewBinding/app/src/main/java/com/incava/ex78viewbinding/MyAdapter.java
package com.incava.ex78viewbinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.incava.ex78viewbinding.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    Context context;
    ArrayList<ItemVO> items;

    public MyAdapter(Context context, ArrayList<ItemVO> items) {
        this.context = context;
        this.items = items;
    }


    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(context).inflate(R.layout.recycler_item,parent,false);
        return new VH(itemView);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        holder.binding.tv.setText(items.get(position).title);
        holder.binding.iv.setImageResource(items.get(position).imgResId);
    }

    class VH extends RecyclerView.ViewHolder{

        RecyclerItemBinding binding;
        public VH(@NonNull View itemView) {
            super(itemView);
            binding = RecyclerItemBinding.bind(itemView);//이미 바인딩이 되어있으므로 bind.

        }
    }


}

File: Ex82HttpRequest/app/src/main/java/com/incava/ex82httprequest/MyAdapter.kt
package com.incava.ex82httprequest

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.incava.ex82httprequest.databinding.RecyclerItemBinding

class MyAdapter(list : MutableList<ItemVO> ) : RecyclerView.Adapter<MyAdapter.VH>() {
    private var items : MutableList<ItemVO>
    init{
        items = list
    }

    inner class VH(private val binding: RecyclerItemBinding) : ViewHolder(binding.root) {
        fun bind(itemVO: ItemVO){
            binding.apply {
                tvNo.text = itemVO.no.toString()
                tvMsg.text = itemVO.msg
                tvDate.text = itemVO.date
            }
        }
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        var binding = RecyclerItemBinding.inflate(LayoutInflater.from(parent.context),parent,false)
        return VH(binding)
    }

    override fun getItemCount(): Int {
       return items.size
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.bind(items[position])
    }


}